File: rssparser/src/main/java/com/prof/rssparser/Article.java
/*
 * The MIT License (MIT)
 *
 *    Copyright (c) 2016 Marco Gomiero
 *
 *    Permission is hereby granted, free of charge, to any person obtaining a copy
 *    of this software and associated documentation files (the "Software"), to deal
 *    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *    copies of the Software, and to permit persons to whom the Software is
 *    furnished to do so, subject to the following conditions:
 *
 *    The above copyright notice and this permission notice shall be included in all
 *    copies or substantial portions of the Software.
 *
 *    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 *    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *    SOFTWARE.
 *
 */

package com.prof.rssparser;

import java.util.Date;

/**
 * Created by Marco Gomiero on 12/02/2015.
 */
public class Article {

    private String title;
    private String author;
    private String link;
    private Date pubDate;
    private String description;
    private String content;
    private String image;

    public String getTitle() {

        return title;

    }

    public String getAuthor() {

        return author;

    }

    public String getLink() {

        return link;

    }

    public Date getPubDate() {

        return pubDate;

    }

    public String getDescription() {

        return description;

    }

    public String getContent() {

        return content;

    }

    public String getImage() {

        return image;

    }

    public void setTitle(String title) {

        this.title = title;

    }

    public void setAuthor(String author) {

        this.author = author;

    }

    public void setLink(String link) {

        this.link = link;

    }

     public void setPubDate(Date pubDate) {

        this.pubDate = pubDate;

    }

    public void setDescription(String description) {

        this.description = description;

    }

    public void setContent(String content) {

        this.content = content;

    }

    public void setImage(String image) {

        this.image = image;

    }

    @Override
    public String toString() {

        return "Article{" +
                "title='" + title + '\'' +
                ", author='" + author + '\'' +
                ", link='" + link + '\'' +
                ", pubDate=" + pubDate +
                ", description='" + description + '\'' +
                ", content='" + content + '\'' +
                ", image='" + image + '\'' +
                '}';

    }

}

File: rssparser/src/main/java/com/prof/rssparser/Article.kt
/*
*   Copyright 2016 Marco Gomiero
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*
*/

package com.prof.rssparser

data class Article(
        var title: String? = null,
        var author: String? = null,
        var link: String? = null,
        var pubDate: String? = null,
        var description: String? = null,
        var content: String? = null,
        var image: String? = null,
        private var _categories: MutableList<String> = mutableListOf()
) {

    val categories: MutableList<String>
        get() = _categories

    fun addCategory(category: String) {
        _categories.add(category)
    }
}